pkg/registry/apis/provisioning/repository_controller.go

```diff
@@ -334,44 +322,65 @@ func (rc *RepositoryController) process(item *queueItem) error {
 			}
 		}
 
-		if !res.Success {
-			logger.Error("repository is unhealthy", "errors", res.Errors)
-		}
-
 		status.Health = provisioning.HealthStatus{
-			Checked: status.Health.Checked,
 			Healthy: res.Success,
+			Checked: time.Now().UnixMilli(),
 			Message: res.Errors,
 		}
+		if !res.Success {
+			logger.Error("repository is unhealthy", "errors", res.Errors)
+		}
 	}
 
-	cfg := cachedRepo.DeepCopy()
-	cfg.Status = *status
-	if _, err := rc.client.Repositories(cachedRepo.GetNamespace()).
-		UpdateStatus(ctx, cfg, v1.UpdateOptions{}); err != nil {
-		return fmt.Errorf("update status: %w", err)
+	// Queue a sync job
+	hash := status.Sync.Hash
+	if status.Sync.State == provisioning.JobStateWorking {
+		res := rc.jobs.Status(ctx, obj.Namespace, status.Sync.JobID)
```

Comment:
I am not sure your motivation for this change. 

the status update should be part of the queue and the job. Not in the controller. The controller and other parts of the code should simply submit.

Reply:
I kept hitting this while testing and it could mess things up -- it is mostly important for the startup cycle where everything in sent to the controller (eg, not really an "update")

For things that are "working" we make sure they are actually still working

Reply:
ok, let's keep it for now. I will audit the controller in the polishing phase